src/server/api/controllers/products.controller.js

```diff
@@ -78,19 +57,48 @@ const getProductById = async (id) => {
         'is_on_discount',
         'discount_percent',
         'picture',
-      )
-      .where({ id });
-    if (products.length === 0) {
-      throw new Error(`incorrect entry with the id of ${id}`, 404);
+    )
+    .where('products.is_on_discount', '=', '1');
+
+const getProductById = async(prod_id) => {
+    const id = parseInt(prod_id, 10);
+    if (isNaN(prod_id) || id < 1) {
+        throw new HttpError('Bad request, Invalid id', 400);
+    } else {
```

Comment:
Bonus:
This `else` itself is not needed, the `if` above works as a [guard](https://betterprogramming.pub/refactoring-guard-clauses-2ceeaa1a9da) by throwing.

Prefer instead code like this:
```js
if (isNaN(prod_id) || id < 1) {
    throw new HttpError('Bad request, Invalid id', 400);
}

try {
 ...
```